src/Dfe.PlanTech.Infrastructure.Contentful/Persistence/ContentfulRepository.cs

```diff
@@ -56,10 +60,21 @@ private static string CreateErrorString(ContentfulError error)
     }
 
     public async Task<IEnumerable<TEntity>> GetEntities<TEntity>(CancellationToken cancellationToken = default)
-    => await GetEntities<TEntity>(LowerCaseFirstLetter(typeof(TEntity).Name), null, cancellationToken);
+    {
+        var contentType = LowerCaseFirstLetter(typeof(TEntity).Name);
+        var key = $"{contentType}s";
+
+        return await _cache.GetOrCreateAsync(key, async () => await GetEntities<TEntity>(contentType, null, cancellationToken)) ?? [];
+    }
 
     public async Task<IEnumerable<TEntity>> GetEntities<TEntity>(IGetEntitiesOptions options, CancellationToken cancellationToken = default)
-        => await GetEntities<TEntity>(LowerCaseFirstLetter(typeof(TEntity).Name), options, cancellationToken);
+    {
+        var contentType = LowerCaseFirstLetter(typeof(TEntity).Name);
+        var jsonOptions = JsonConvert.SerializeObject(options);
```

Comment:
Generally it's preferred to use `System.Text.Json` instead of `Newtonsoft.Json` where possible. _But_ the Contentful Nuget package already uses the latter so it's not like you're adding it here or anything.

Reply:
Ah thank you - I'll use that if I put it back, but for now its gone re comment about keys below